### DISCUSSION TITLE
Serverless Function timeout in project with no serverless functions

### DISCUSSION THREAD

**Discussion by mathealgou** [vercel/community#2594] [Help] [answered]
Hi, I have a Next.Js project hosted on Vercel using Next's version 13.3.1 using /app in which one specific page returns the following:

![image](https://github.com/vercel/community/assets/23152411/e8757b39-5107-40e6-bca7-85db1d0f229d)

There are no serverless functions in the project, so I would assume it has something to do with the deployment. The page itself is nothing special, and in my tests the one backend call it does to an external service always returns 200, even when the page breaks.

What could cause this?

**Comment by amyegan** [comment]
Hi @mathealgou! If you share your public repo or a minimal reproduction, I can take a closer look into the source of the error.

**Reply by mathealgou** [reply]
While I can't give the exact code, it's a pretty basic search result page, all it does is make an API call to validate a token and one to get the results, there is error handling in the API calls, but while watching the logs in the backend, I haven't been able to spot any errors. It's important to point out that the back end does not live in the same project and it's in an entirely different programming language. 

Also, the components used in the page are also used in other pages, where this error doesn't occur. I'm led to believe this might me a problem with Vercel's deployment rather than the specific code.

**Reply by amyegan** [reply, marked as answer]
I know the new App Router uses React [Server Components](https://nextjs.org/docs/getting-started/react-essentials#server-components), so it's possible the page has a serverless function depending on how it is configured. Or does the page call the separate backend or another service? 

The [FUNCTION_INVOCATION_TIMEOUT](https://vercel.com/docs/errors/application-error-codes#function_invocation_timeout) error means that a request reached the [timeout threshold](https://vercel.com/docs/concepts/limits/overview#serverless-function-execution-timeout). A common cause for this issue is when an environment variable is misconfigured, so I recommend double checking env vars as well.

**Reply by mathealgou** [reply]
Have been able to "unclassify" the code, here it is:

```typescript
// * Services
import { search } from "@/services";
import { verifyToken } from "@/services/auth";

// * Hooks and Utils
import { cookies } from "next/headers";
import { redirect } from "next/navigation";

// * Components
import ArticleCard from "@/components/articleCard";
import QuestionCard from "@/components/questionCard";
import Pagination from "@/components/pagination";

interface Props {
	params?: {
		searchTerm?: string;
	};
	searchParams?: {
		page?: number;
	};
}

export default async function Page(props: Props) {
	const cookieStore = cookies();
	const token = cookieStore.get("@App:token:access");
	const isValid = await verifyToken(token?.value);

	const page = props?.searchParams?.page || 1;

	if (!isValid) {
		redirect(`/auth/login?callbackUrl=/search/${props?.params?.searchTerm}`);
	}

	const term = decodeURIComponent(props?.params?.searchTerm || "");

	const { articles, questions, count } = await search(term, page, token?.value);

	return (
		<main>
			<div>
				<section>
					<div>
						{articles.map((article: any) => (
							<ArticleCard key={article.id} article={article} />
						))}
					</div>
					<div>
						{questions.map((question: any) => (
							<QuestionCard key={question.id} question={question} />
						))}
					</div>
				</section>
			</div>
			<div>
				<Pagination />
			</div>
		</main>
	);
}
```

It does call a separate backend, in the `search()` and `verify()` function calls, but I think it's unlikely that's causing a problem since this page hasn't been changed in the last 2 weeks or so, and the error just started appearing the last two days. Also, this has been happening in a preview environment, so the ping to the backend shouldn't be a problem.
